### PR TITLE
Add include_blob_files option to GetApproximateSizes

### PR DESCRIPTION
**Pull request by xingbowang**
**Summary:**
Add a new boolean flag `include_blob_files` (default: `false`) to `SizeApproximationOptions` and a corresponding `INCLUDE_BLOB_FILES` enum value to `SizeApproximationFlags`. When set to `true`, the returned size includes an approximation of blob file data in the queried key range.

**Algorithm:**
The blob file size contribution is prorated using the SST size ratio:
```
blob_size_in_range ≈ total_blob_size * (sst_size_in_range / total_sst_size)
```
The blob-to-SST ratio (`total_blob_size / total_sst_size`) is computed once before the per-range loop, so iterating levels and blob files only happens once per `GetApproximateSizes` call regardless of how many ranges are queried. The per-range SST size (`ApproximateSize`) is computed once and shared between `include_files` and `include_blob_files`.

**Limitations:**
- Assumes blob data is distributed proportionally to SST data across the key space. May be inaccurate if blob value sizes vary significantly across different key ranges (e.g., one range has large blobs while another has small ones).
- If there are no SST files (all data in memtables), the blob size contribution will be 0 even if blob files exist on disk.

**Changes:**
- `include/rocksdb/options.h`: New `include_blob_files` field in `SizeApproximationOptions`; updated doc comments for `include_memtables`/`include_files`
- `include/rocksdb/db.h`: New `INCLUDE_BLOB_FILES` in `SizeApproximationFlags` enum, updated flags-to-options mapping
- `include/rocksdb/c.h`: New `rocksdb_size_approximation_flags_include_blob_files` C API enum value
- `java/`: Added `INCLUDE_BLOB_FILES` to `SizeApproximationFlag.java` and JNI flag mapping in `rocksjni.cc`
- `db/db_impl/db_impl.cc`: Blob-to-SST ratio computed once before loop, SST range size computed once per range and shared
- `db_stress_tool/db_stress_test_base.cc`: Randomized `include_blob_files` in stress test

**Test Plan:**
- New `DBBlobBasicTest.GetApproximateSizesIncludingBlobFiles` — verifies:
  - Size with blobs > without (full range)
  - Non-overlapping range returns 0
  - Partial range returns proportionally less than full range
  - `SizeApproximationFlags` API works
  - Multi-range query: two sub-ranges sum approximately to the full-range result
- Stress test now exercises the new option randomly

### CODE REVIEW

File: .claude/review-8bcfa9c5d-enforce-wbm-recovery/consensus.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
+# Consensus Document: enforce_write_buffer_manager_during_recovery Review
```
Comment (xingbowang) [inline .claude/review-8bcfa9c5d-enforce-wbm-recovery/consensus.md]: remove these files in .claude folder. they are from previous work.

File: flush_error_analysis.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
+# RocksDB Flush Failure, Background Error, and Auto-Recovery Analysis
```
Comment (xingbowang) [inline flush_error_analysis.md]: delete these .md file as well

File: db/version_set.cc [hunk 3]
```diff
@@ -7539,6 +7539,56 @@ uint64_t VersionSet::ApproximateSize(const ReadOptions& read_options,
                                       *f.file_metadata, caller, icmp, cf_opts);
 }
 
+uint64_t VersionSet::ApproximateBlobSize(
```
Comment (xingbowang) [inline db/version_set.cc]: We are duplicating the logic in ApproximateSize, could we extract some common function to dedup the logic?

**Comment by xingbowang** [comment]
The CI failure in `build-linux-cmake-with-folly-coroutines` is the flaky `EventListenerTest.BackgroundJobPressure` test, unrelated to this PR. It was introduced in #14474 and "fixed" in #14476, but is still intermittently failing. Our previous push (commit b8dcb3b) passed this same job; the failure only appeared on the latest push with the same test code.

File: .claude/review-8bcfa9c5d-enforce-wbm-recovery/consensus.md (hunk 1 above)
Comment (xingbowang) [reply .claude/review-8bcfa9c5d-enforce-wbm-recovery/consensus.md]: Done — removed .claude/ directory.

File: flush_error_analysis.md (hunk 2 above)
Comment (xingbowang) [reply flush_error_analysis.md]: Done — removed stray .md files.

File: db/version_set.cc (hunk 3 above)
Comment (xingbowang) [reply db/version_set.cc]: Done — removed `ApproximateBlobSize()` entirely. The blob-to-SST ratio (`total_blob_size / total_sst_size`) is now computed once before the per-range loop in `DBImpl::GetApproximateSizes`, and the per-range SST size from `ApproximateSize()` is computed once and shared between `include_files` and `include_blob_files`. No duplicated logic.

**Comment by xingbowang** [comment]
Addressed the following from the Claude Code Review:

**#1 (Critical)** — Updated doc comments for `include_memtables` and `include_files` to reflect `include_blob_files` as a third valid option.

**#3 (C API / Java JNI)** — Added `rocksdb_size_approximation_flags_include_blob_files` to C API enum (`c.h`), `INCLUDE_BLOB_FILES` to Java `SizeApproximationFlag.java`, and JNI flag mapping in `rocksjni.cc`.

**#4 (Stress test)** — Added `sao.include_blob_files = thread->rand.OneIn(2)` to the stress test's `GetApproximateSizes` path.

**#5 (Performance)** — Hoisted `total_sst_size` and `total_blob_size` computation out of the per-range loop. The blob-to-SST ratio is now computed once per `GetApproximateSizes` call. Also removed the separate `VersionSet::ApproximateBlobSize()` method — the logic is inlined in `DBImpl::GetApproximateSizes`.

**#7 (const)** — Moot since the method was removed entirely; the inlined code only reads from the `Version*`.

**Intentionally not addressed:**
- **#2 (Garbage bytes)** — Kept as-is to match existing SST behavior (raw file sizes, not live data).
- **#6 (Floating-point >9PB)** — Impractical edge case; double precision is sufficient.
- **#8 (Test nitpick)** — Minor; the check works correctly for realistic file names.
- **#9 (HISTORY.md)** — Will add before merge.

File: db/blob/db_blob_basic_test.cc [hunk 4]
```diff
@@ -2459,6 +2459,111 @@ TEST_F(DBBlobWithTimestampTest, IterateBlobs) {
...
+      has_blob_files = true;
+      break;
+    }
+  }
+  ASSERT_TRUE(has_blob_files);
+
+  // Query the full range - all keys are covered.
+  std::string start = Key(0);
+  std::string end = Key(kNumKeys);
+  Range r(start, end);
```
Comment (hx235) [inline db/blob/db_blob_basic_test.cc:2494]: Seem like both stress test and unit test only cover one single range, may worth twisting either of them to cover two ranges at least.

**Review by hx235 — APPROVED** [review]
LGTM!

**Comment by hx235** [comment]
You probably need to reimport so that I can stamp internally

File: db/blob/db_blob_basic_test.cc (hunk 4 above)
Comment (xingbowang) [reply db/blob/db_blob_basic_test.cc:2494]: Done — added a multi-range test case that queries two non-overlapping sub-ranges and verifies both return positive sizes, and that their sum is approximately equal to the full-range result (within 10%).
